src/player/process-death.cpp

```diff
@@ -42,172 +42,155 @@
  */
-static void show_basic_params(PlayerType *player_ptr, char *buf)
+static void show_basic_params(PlayerType *player_ptr)
 {
-    char tomb_message[160];
-    (void)sprintf(tomb_message, _("レベル: %d", "Level: %d"), (int)player_ptr->lev);
-    center_string(buf, tomb_message);
-    put_str(buf, 11, 11);
-
-    (void)sprintf(tomb_message, _("経験値: %ld", "Exp: %ld"), (long)player_ptr->exp);
-    center_string(buf, tomb_message);
-    put_str(buf, 12, 11);
-
-    (void)sprintf(tomb_message, _("所持金: %ld", "AU: %ld"), (long)player_ptr->au);
-    center_string(buf, tomb_message);
-    put_str(buf, 13, 11);
+    put_str(center_string(format(_("レベル: %d", "Level: %d"), (int)player_ptr->lev)).data(), 11, 11);
+
+    put_str(center_string(format(_("経験値: %ld", "Exp: %ld"), (long)player_ptr->exp)).data(), 12, 11);
+
+    put_str(center_string(format(_("所持金: %ld", "AU: %ld"), (long)player_ptr->au)).data(), 13, 11);
 }
 
 #ifdef JP
 /*!
  * @brief プレイヤーを殺したモンスターを表示する (日本語版専用)
  * @param player_ptr プレイヤーへの参照ポインタ
- * @param buf 墓テンプレ
- * @param tomb_message 墓碑に刻む言葉
- * @return 追加の行数
+ * @return メッセージと余分な行数のペア
  */
-static int show_killing_monster(PlayerType *player_ptr, char *buf, char *tomb_message, size_t tomb_message_size)
+static std::pair<std::string, int> show_killing_monster(PlayerType *player_ptr)
 {
-    shape_buffer(player_ptr->died_from.data(), GRAVE_LINE_WIDTH + 1, tomb_message, tomb_message_size);
-    char *t;
-    t = tomb_message + strlen(tomb_message) + 1;
+    char buf[160];
+    shape_buffer(player_ptr->died_from.data(), GRAVE_LINE_WIDTH + 1, buf, sizeof(buf));
+    char *t = buf + strlen(buf) + 1;
     if (!*t) {
-        return 0;
+        return std::pair<std::string, int>(buf, 0);
     }
 
-    char killer[MAX_MONSTER_NAME];
-    strcpy(killer, t); /* 2nd line */
+    std::string killer = t; /* 2nd line */
     if (*(t + strlen(t) + 1)) /* Does 3rd line exist? */
     {
-        for (t = killer + strlen(killer) - 2; iskanji(*(t - 1)); t--) { /* Loop */
-            ;
-        }
-        strcpy(t, "…");
-    } else if (angband_strstr(tomb_message, "『") && suffix(killer, "』")) {
-        char killer2[MAX_MONSTER_NAME];
-        char *name_head = angband_strstr(tomb_message, "『");
-        sprintf(killer2, "%s%s", name_head, killer);
-        if (strlen(killer2) <= GRAVE_LINE_WIDTH) {
-            strcpy(killer, killer2);
-            *name_head = '\0';
+        auto i = killer.length() - 2;
+        while (iskanji(killer[i - 1])) {
+            --i;
         }
-    } else if (angband_strstr(tomb_message, "「") && suffix(killer, "」")) {
-        char killer2[MAX_MONSTER_NAME];
-        char *name_head = angband_strstr(tomb_message, "「");
-        sprintf(killer2, "%s%s", name_head, killer);
-        if (strlen(killer2) <= GRAVE_LINE_WIDTH) {
-            strcpy(killer, killer2);
-            *name_head = '\0';
+        killer.erase(i);
+        killer.append("…");
+    } else {
+        char *name_head = angband_strstr(buf, "『");
```

Comment:
It is better to keep the original indentation structure for readability.
If you are concerned about the inefficiency of running angband_strstr twice, you can use the following initialization syntax within an if statement.

```c++
        } else if (auto *name_head = angband_strstr(buf, "『");
            name_head && suffix(killer.data(), "』"))) {
            std::string killer2 = name_head;
            killer2.append(killer);
            if (killer2.length() <= GRAVE_LINE_WIDTH) {
                killer = killer2;
                *name_head = '\0';
            }
        } else if (...) {
        }
```

Reply:
The forced push reverted to the indentation of those two else clauses in the existing code.